json/store.go

```diff
@@ -0,0 +1,126 @@
+package json
+
+import (
+	"encoding/json"
+	"fmt"
+	"go.mozilla.org/sops"
+	"go.mozilla.org/sops/kms"
+	"go.mozilla.org/sops/pgp"
+	"time"
+)
+
+type JSONStore struct {
+}
+
+func (store JSONStore) Load(in string) (sops.TreeBranch, error) {
+	var branch sops.TreeBranch
+	err := json.Unmarshal([]byte(in), branch)
+	if err != nil {
+		return branch, fmt.Errorf("Could not unmarshal input data: %s", err)
+	}
+	for i, item := range branch {
+		if item.Key == "sops" {
+			branch = append(branch[:i], branch[i+1:]...)
+		}
+	}
+	return branch, nil
+}
+
+func (store JSONStore) Dump(tree sops.TreeBranch) (string, error) {
+	out, err := json.Marshal(tree)
+	if err != nil {
+		return "", fmt.Errorf("Error marshaling to json: %s", err)
+	}
+	return string(out), nil
+}
+
+func (store JSONStore) DumpWithMetadata(tree sops.TreeBranch, metadata sops.Metadata) (string, error) {
+	tree = append(tree, sops.TreeItem{Key: "sops", Value: metadata.ToMap()})
+	out, err := json.Marshal(tree)
+	if err != nil {
+		return "", fmt.Errorf("Error marshaling to json: %s", err)
+	}
+	return string(out), nil
+}
+
+func (store JSONStore) LoadMetadata(in string) (sops.Metadata, error) {
+	var metadata sops.Metadata
+	data := make(map[interface{}]interface{})
+	encoded := make(map[interface{}]interface{})
+	if err := json.Unmarshal([]byte(in), &encoded); err != nil {
+		return metadata, fmt.Errorf("Error unmarshalling input json: %s", err)
+	}
+
+	sopsJSON, err := json.Marshal(encoded["sops"])
+	if err != nil {
+		return metadata, err
+	}
+
+	err = json.Unmarshal(sopsJSON, &data)
+	if err != nil {
+		return metadata, err
+	}
```

Comment:
Uh... Good question. That basically grew from https://github.com/mozilla/sops/blob/go-sops/decrypt.go#L12-L49 without much thought on my side. I don't see any reason why we can't just do data = encoded["sops"].(map[interface{}]interface{})